analysis_engine/weight_transfer.py

```
@@ -0,0 +1,177 @@
+def render_weight_graph(
+    result: WeightTransferResult, impact: Optional[ImpactInfo], path: str
+) -> Optional[str]:
+    """Render the weight-transfer / balance movement graph as a PNG."""
+    if not result.series:
+        return None
+    try:
+        import matplotlib
+
+        matplotlib.use("Agg")
+        import matplotlib.pyplot as plt
+    except Exception:  # pragma: no cover - optional dependency
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)